src/components/form/formComponent.jsx

```diff
@@ -94,11 +96,27 @@ export function createForm(config) {
                     ...currentState,
                     ...(isFunction(stateUpdate) ? stateUpdate(currentState) : stateUpdate)
                 };
-                const validation = this.validateField(this.fieldByKey[fieldName], newState, prevState);
                 const newFields = {
                     ...prevState.fields,
                     [fieldName]: {
                         ...newState,
+                    }
+                };
+
+                return {
+                    fields: newFields
+                };
+            });
+        };
+
+        handleFieldValidation = (fieldName) => {
+            this.setState(prevState => {
+                const currentState = prevState.fields[fieldName];
+                const validation = this.validateField(this.fieldByKey[fieldName], currentState, this.state);
+                const newFields = {
+                    ...prevState.fields,
```

Comment:
You don't need to reproduce the entire previous state while using 'setState'.

Reply:
Here you have to, as you're modifying one field in the array of fields, which you're entirely replacing on line 126.